app/moderation/page.tsx

```diff
@@ -12,8 +20,483 @@ export async function generateMetadata(): Promise<Metadata> {
+          FROM users_with_limits uwl
+          LEFT JOIN "Users" u ON uwl."userId" = u._id
+          ${!showNewUserRateLimits ? `WHERE COALESCE(u."postCount", 0) + COALESCE(u."commentCount", 0) >= 5` : ''}
+          ORDER BY uwl."mostRecentActivation" DESC
+          LIMIT $1 OFFSET $2
+        `, [limit, activeRateLimitsOffset]),
+        db.one<{count: number}>(`
+          WITH latest_events AS (
+            SELECT DISTINCT ON (e."userId", e.properties->>'rateLimitType', e.properties->>'actionType')
+              e."userId",
+              e.name,
+              e.properties,
+              COALESCE((e.properties->>'triggeredAt')::timestamp, e."createdAt") as "triggeredAt"
+            FROM "LWEvents" e
+            WHERE e.name IN ('rateLimitActivated', 'rateLimitDeactivated')
+            ORDER BY e."userId", e.properties->>'rateLimitType', e.properties->>'actionType', e."createdAt" DESC
+          ),
+          filtered_limits AS (
+            SELECT DISTINCT le."userId"
+            FROM latest_events le
+            WHERE
+              ${showExpiredRateLimits
+                ? `le.name = 'rateLimitDeactivated'`
+                : `le.name = 'rateLimitActivated'`
+              }
+          )
+          SELECT COUNT(*) as count
+          FROM filtered_limits fl
+          ${!showNewUserRateLimits ? `
+            LEFT JOIN "Users" u ON fl."userId" = u._id
+            WHERE COALESCE(u."postCount", 0) + COALESCE(u."commentCount", 0) >= 5
+          ` : ''}
+        `)
+      ]);
+
+  const activeRateLimits = activeRateLimitsData.map(row => ({
+    userId: row.userId,
+    rateLimits: JSON.parse(row.rateLimits),
+    mostRecentActivation: row.mostRecentActivation,
+    user: row.user__id ? {
+      _id: row.user__id,
+      displayName: row.user_displayName,
+      slug: row.user_slug,
+      createdAt: row.user_createdAt,
+      karma: row.user_karma,
+      postCount: row.user_postCount,
+      commentCount: row.user_commentCount,
+    } : null,
+  }));
+  const activeRateLimitsCount = activeRateLimitsCountResult.count;
+
+  // Fetch rejected posts
+  const [rejectedPostsData, rejectedPostsCountResult] = await Promise.all([
+        db.manyOrNone<DbRow>(`
+          SELECT
+            p._id, p.title, p.slug, p."userId", p."postedAt", p."rejectedReason",
+            u._id as "user__id", u."displayName" as "user_displayName", u.slug as "user_slug",
+            ru._id as "reviewedByUser__id", ru."displayName" as "reviewedByUser_displayName", ru.slug as "reviewedByUser_slug"
+          FROM "Posts" p
+          LEFT JOIN "Users" u ON p."userId" = u._id
+          LEFT JOIN "Users" ru ON p."reviewedByUserId" = ru._id
+          WHERE p.rejected = TRUE
+          ORDER BY p."postedAt" DESC NULLS LAST
+          LIMIT $1 OFFSET $2
+        `, [limit, rejectedPostsOffset]),
+        db.one<{count: number}>(`
+          SELECT COUNT(*) as count
+          FROM "Posts"
+          WHERE rejected = TRUE
```

Comment:
I think by default prefer `IS` for comparing booleans, since comparisons to null return always `false` (which is correct in this case but not in all cases)